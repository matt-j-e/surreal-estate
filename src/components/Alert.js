import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Success = styled.p`
  padding: 1rem;
  text-align: center;
  color: mediumseagreen;
  border: 2px solid mediumseagreen;
  font-weight: 900;
`;

const Error = styled.div`
  padding: 1rem;
  text-align: center;
  color: palevioletred;
  border: 2px solid palevioletred;
  font-weight: 900;
`;

const Alert = ({ message, success }) => {
  if (!message) return "";
  return success ? <Success>{message}</Success> : <Error>{message}</Error>;
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  success: false,
};

export default Alert;
