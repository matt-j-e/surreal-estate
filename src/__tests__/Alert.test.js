import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("renders an 'Error' alert", () => {
    render(<Alert message="Error!" />);
    const alertMessage = screen.getByText(/Error!/);
    expect(alertMessage).toBeInTheDocument();
  });

  it("renders a 'Success' alert", () => {
    render(<Alert message="Success!!!" success />);
    const alertMessage = screen.getByText(/Success!!!/);
    expect(alertMessage).toBeInTheDocument();
  });

  it("renders nothing when no message to show", () => {
    render(<Alert message="" />);
    const successMessage = screen.queryByText(
      /Your property was added to the database/
    );
    const errorMessage = screen.queryByText(
      /A server error occurred. Please try again./
    );
    expect(successMessage).not.toBeInTheDocument();
    expect(errorMessage).not.toBeInTheDocument();
  });
});

describe("Alert rendering", () => {
  it("renders the error Alert correctly", () => {
    const { asFragment } = render(
      <Alert message="A server error occurred. Please try again." />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the success Alert correctly", () => {
    const { asFragment } = render(
      <Alert message="Your property was added to the database." success />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders no alert when no message is passed in", () => {
    const { asFragment } = render(<Alert message="" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
