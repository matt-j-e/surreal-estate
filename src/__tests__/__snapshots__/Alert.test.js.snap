// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert rendering renders no alert when no message is passed in 1`] = `<DocumentFragment />`;

exports[`Alert rendering renders the error Alert correctly 1`] = `
<DocumentFragment>
  <p
    class="sc-gsDJrp jSTqYq"
  >
    A server error occurred. Please try again.
  </p>
</DocumentFragment>
`;

exports[`Alert rendering renders the success Alert correctly 1`] = `
<DocumentFragment>
  <p
    class="sc-bdvvaa feCLug"
  >
    Your property was added to the database.
  </p>
</DocumentFragment>
`;
